compiler/src/main/kotlin/dev/zacsweers/metro/compiler/fir/generators/ContributedInterfaceSupertypeGenerator.kt

```diff
@@ -118,6 +119,20 @@ internal class ContributedInterfaceSupertypeGenerator(session: FirSession) :
       }
     }
 
+  private fun FirRegularClassSymbol.contributesAnnotations(): Sequence<FirAnnotation> {
+    return annotations.annotationsIn(session, session.classIds.allContributesAnnotations).ifEmpty {
+      // When there are multiple, they end up getting bundled into a 'Container' type that we need
+      // to unwrap. This behavior appears to only occur for contributions from other
+      // modules/compilations.
+      ((annotations.firstOrNull()?.argumentMapping?.mapping?.values?.firstOrNull()
+            as? FirArrayLiteral)
+          ?.argumentList
+          ?.arguments as? List<FirAnnotation>)
```

Comment:
I feel like there's probably a better way to do this but I didn't see anything else that worked. @bnorm do you know of any helpers that might be useful here?

Reply:
I did add an `.expectAs()` helper to at least save some of the parens 🥲

Reply:
The chain simultaneously looks better and crazier now 😂

Reply:
There are a couple in the compiler I found; `unwrapAndFlattenArgument` and `unwrapVarargValue` for example. But I think it might be better for compatibility with future Kotlin compiler versions if you write your own utility using those as inspiration.